### ISSUE TITLE
Models not registered in admin with AdminSiteOTPRequired?

### ISSUE THREAD

**Issue by bctiemann** [Bouke/django-two-factor-auth#138]
Hi--having some trouble here.

I'm trying to add two-factor auth to my site's admin. I've got this in my urls.py:

```
otp_admin_site = AdminSiteOTPRequired()

urlpatterns = [
    url(r'', include('two_factor.urls', 'two_factor')),
    url(r'^admin/', include(otp_admin_site.urls)),
    ...
]
```

And it logs me in fine, but then I see "You don't have permission to edit anything." in the admin interface. (Normally, with `include(admin.site.urls)`, I get full access to all the models.)

Seems like the models aren't being registered? What could be causing this?

**Comment by bctiemann** [comment]
After further investigation I see that I have to do this in the urls.py to explicitly register all my models with the new otp_admin_site object:

`otp_admin_site.register(Widget)`

But I'm guessing this means the usual convention of keeping ModelAdmin instances in admin.py isn't supported, nor is the `@register` decorator (for making WidgetAdmin instances with special configurations)?

**Comment by bctiemann** [comment]
I was able to get this working, including @register. I think this is just a documentation matter if anything.

rather than simply pointing people to AdminSiteOTPRequired and AdminSiteOTPRequiredMixin, maybe just an example admin.py:

```
from django.contrib.auth.models import User, Group
from django.contrib.auth.admin import UserAdmin, GroupAdmin
from two_factor.admin import AdminSiteOTPRequired, admin

otp_admin_site = AdminSiteOTPRequired()

# These are not automatically registered as they are with standard admin
otp_admin_site.register(User, UserAdmin)
otp_admin_site.register(Group, GroupAdmin)

@admin.register(Widget, site=otp_admin_site)
class WidgetAdmin(admin.ModelAdmin):
    list_display = ('id', 'name', 'url',)
```

**Comment by Bouke** [comment]
By default, the admin site is patched to use two-factor authentication. If you don't like the patching, but still want to enable two-factor authentication for the admin site, you have to use a different admin site. The documentation of Django should be sufficient in that regard.

**Comment by bctiemann** [comment]
Thanks for the update, but I don't think I see how this addresses the source of my confusion, which was that if you accept the patching (which is fine!), none of your ModelAdmins get registered anymore, and you have to do it manually. That goes for all the built-in models (User, Group), app models, and models from django_two_factor_auth itself, which you have to dig up and explicitly register.

**Comment by Bouke** [comment]
Please refer to the example app as part of this package; it uses the patching, doesn't register the django.contrib.auth models, and those are still available in the admin. So I think there's something wrong with your configuration.

**Comment by bctiemann** [comment]
Hi--I did some more digging on this, and I'm starting to think that on my installation at least the patching isn't actually doing anything.

Using the example app, in two_factor/admin.py, I changed patch_admin to this:

def patch_admin():
    print 'patching'
    @monkeypatch_method(AdminSite)
    def login(self, request, extra_context=None):
        print 'new login'
        return None

And it prints "patching" to the console, but the login method that executes is still the original login() from contrib.admin. It never reaches the "new login" output.

I need to experiment more to see why this is happening, but that's the root of the issue.
